content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+```{r, lp-sign}
+fit_censored <- proportional_hazards() %>% 
+  set_engine("survival") %>% 
+  fit(Surv(time, status) ~ X1 + X2, data = train_dat)
+pred_censored <- predict(fit_censored, type = "linear_pred", new_data = test_pred)
+
+pred <- bind_cols(
+  pred,
+  pred_censored %>% rename(pred_censored = .pred_linear_pred)
+)
+
+pred %>% 
+  ggplot(aes(linear_pred, pred_censored)) + 
+  geom_point() +
+  geom_abline(slope = -1, intercept = 0)
+```
+
+  
+
+### Model and prediction types available in censored
+
+Currently, censored contains the following models, engine, and prediction types with `r cli::symbol$tick` indicating "available", `r cli::symbol$cross` indicating "unavailable", and `r cli::symbol$circle` indicating "to do".
+
+```{r table, echo = FALSE, results = "asis"}
+yep <- cli::symbol$tick
+nope <- cli::symbol$cross
+todo <- cli::symbol$circle
+
+mod_names <- get_from_env("models")
+model_info <-
+  map_dfr(mod_names, ~ get_from_env(paste0(.x, "_predict")) %>% mutate(alias = .x))
+
+done <- model_info %>%
+  filter(mode == "censored regression") %>%
+  select(model = alias, engine, type, mode) %>% 
+  pivot_wider(names_from = type, 
+              values_from = mode, 
+              values_fill = nope, 
+              values_fn = function(x) yep) %>% 
+  mutate(linear_pred = case_when(model == "survival_reg" ~ todo,
+                                 TRUE ~ linear_pred),
+         time = case_when(time == nope ~ todo,
+                          TRUE ~ time)
+  )
+to_do <- tribble(
+  ~model, ~engine, ~survival, ~linear_pred, ~time, ~raw, ~quantile, ~hazard,
+  "rand_forest", "ranger", todo, nope, todo, nope, nope, nope,
+  "rand_forest", "randomForestSRC",  todo, nope, todo, nope, nope, nope
+)
+
+bind_rows(done, to_do) %>% 
+  arrange(model, engine) %>% 
+  knitr::kable(escape = FALSE) 
+```
+
+
+
+### What's next?
+
+We want survival analysis to be a first-class citizen in tidymodels which means having access to preprocessing, resampling, models, metrics, and tuning. The next steps on that path are a role for censoring indicator columns and a step `step_surv()` in recipes as well as a new ROC metric for survival models in yardstick. An adaption of workflows and tune is to follow after that. Additionally, we are in the process of figuring out how different types of censoring and time-dependent effects fit into our API.
+
+We'd love to hear from you! Some questions we have include: Would you find this useful? Are there particular models/methods you'd like to see? Do you use time-dependent effects? Do you use interactions? Which types of censoring do you use?
```

Comment:
Same as above. Make them easier to see and link to issues.

Reply:
I think deciding on how to gather feedback is important here. The three main options we talked about in the meeting are:

- using GitHub (a single issue? invite to open an issue?)
- RStudio Community (could use our shortlink to send folks to the right tag/category)
- something like a Google Form that goes to Hannah's inbox

These all have positives/negatives. I think the most important things are that the call-to-action is clear and that Hannah is comfortable with managing it.

Reply:
It's going on RStudio Community!